### ISSUE TITLE
CMake configuration file created by 'make install' does not include Eigen

### ISSUE THREAD

**Issue by remi-moyen** [gstlearn/gstlearn#250]
When generating an installation (`make install`), a set of CMake files are created so that other code can e.g. `find_package(gstlearn)`. This is controlled by `install.cmake`.

However these files do not themselves include Eigen3, despite linking with Eigen3 (or, more properly, using the target "Eigen3::Eigen" in the `INTERFACE_LINK_LIBRARIES`). This means the code that does `find_package(gstlearn)` must itself also do `find_package(Eigen3)`. Failing to do so gives me this error at the end of running CMake (on Windows):
```
Target "foo" links to target "Eigen3::Eigen" but the target was not  found.  Perhaps a find_package() call is missing for an IMPORTED target, or  an ALIAS target is missing?
```

As an aside, one possible way to fix this would be to copy all of Eigen3 directly into the `include/` directory when installing gstlearn (and also setting the relevant paths as needed). This duplicates everything but Eigen3 isn't too big, and it would also have the advantage to ensure that code using Eigen3 would use the same version of Eigen3 as the one used when compiling gstlearn itself. But I think if this was needed, this could be ensured by pegging the Eigen3 version number somewhere in calls to `find_package()`.

What seems a better way is to modify the CMake generated files to properly add this `find_package()`. Following the instructions given [here](https://cmake.org/cmake/help/latest/guide/importing-exporting/index.html) this can be done by:
- Change `install.cmake` to create a `gstlearnTargets.cmake` file instead of `gstlearnConfig.cmake` (I guess any name could be used but this seems to be the common use!).
- Add a call to `configure_package_config_file()` to create the `gstlearnConfig.cmake` file, and `INSTALL()` it.
- Add a template file `Config.cmake.in` that contains essentially just the skeleton given in the link above, plus a `find_dependency(Eigen3)` (the link suggests using `find_dependency()` rather than `find_package()` in this specific case).

I've done those changes and checked that they work, I should be able to send a PR.

**Comment by fabien-ors** [comment]
Thank you Remy.
Nice to here that you should be able to send a PR ;)

**Comment by remi-moyen** [comment]
I haven't yet sent it, let's not be too hasty! ;-)

**Comment by remi-moyen** [comment]
I did not understand how to avoid creating a new issue (#251), I wanted to directly add the PR into this issue. Maybe I'll have more luck next time...

Should I delete the branch I created for the PR once it is merged or should I leave it and simply forget about it?

**Comment by fabien-ors** [comment]
All what you done is OK !
You created a PR #251 (automatically numbered by github) and you referred this Issue #250 in  the comments.
Now, they are linked. All is good :)
Thank you very much

**Comment by pierre-guillou** [comment]
Eigen being a header-only library, I'm not sure why it needs to be exposed like that. Maybe setting the `target_link_libraries(${FLAVOR} PUBLIC Eigen3::Eigen)` to `PRIVATE` might do the trick?

**Comment by pierre-guillou** [comment]
Never mind, Eigen is in fact exposed in gstlearn's public API (as a member of `AMatrixDense`). We indeed need to make gstlearn consumers aware of it through CMake.
